### ISSUE TITLE
Release build broken on macOS?

### ISSUE THREAD

**Issue by andrewvaughanj** [stp/stp#402]
Reported by @carlocab 

```
cmake -S. -Bbuild -DCMAKE_INSTALL_PREFIX=/tmp/stp -DCMAKE_BUILD_TYPE=Release
```

```
cmake --build build
```

```
avj@armadillo /tmp/stp$ clang --version
Apple clang version 12.0.0 (clang-1200.0.32.29)
Target: arm64-apple-darwin20.3.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
avj@armadillo /tmp/stp$ uname -a
Darwin armadillo 20.3.0 Darwin Kernel Version 20.3.0: Thu Jan 21 00:06:51 PST 2021; root:xnu-7195.81.3~1/RELEASE_ARM64_T8101 arm64 arm64 MacBookPro17,1 Darwin
```

```
/tmp/stp/tools/stp/main_common.cpp:342:10: error: no member named 'quick_exit' in namespace 'std'
    std::quick_exit(0);
    ~~~~~^
```

**Comment by msoos** [comment]
Hm, quick_exit is part of C++11 as per [this link](https://www.cplusplus.com/reference/cstdlib/quick_exit/).

It's seemingly `Defined in header <cstdlib>` as per [this link](https://en.cppreference.com/w/cpp/utility/program/quick_exit). Are we missing this header?

Just my 2 cents,

Mate

**Comment by andrewvaughanj** [comment]
Good spot -- it could very well be that (on Linux/GCC), we get `<cstdlib>` via some other header ("accidentally"), but that doesn't hold with `clang` on macOS.

I'll try this later and report back.

**Comment by rgov** [comment]
Is this libc++ (LLVM project) vs libstdc++ (GNU project)?

**Comment by andrewvaughanj** [comment]
`libc++` (macOS 11.3)

**Comment by andrewvaughanj** [comment]
Yeah, @rgov hit the nail on the head: https://stackoverflow.com/a/60496235

I'll add a CMake-based check for this and wrap the quick-exit code based on that.

**Comment by msoos** [comment]
Ah, unimplemented in macOS's libc :rofl: Loving it! Thanks for digging deeper than I did!

**Comment by rgov** [comment]
Do we need to use a quick exit? What is the performance gain? Seems like a lot of complexity to check for availability of an exit call.

**Comment by TrevorHansen** [comment]
> Do we need to use a quick exit? What is the performance gain? 

Generally, I'd guess quick_exit saves about 3% of effort. The saving is from not running the destructors of ASTNodes and hashmap buckets to cleanup their dynamic memory allocations.

Other than checking for availability of quick_exit, I think in the past we used to call exit() directly to get nearly the same effect.

**Comment by msoos** [comment]
Yeah, let's just `exit`. No need for fancy `quick_exit` :smiley: 

Mate

**Comment by rgov** [comment]
Should be fixed by f20c5712
